onnxruntime/core/providers/cuda/shared_inc/cuda_utils.h

```diff
@@ -33,5 +34,25 @@ std::unique_ptr<IConstantBuffer<T>> CreateConstantOnes();
 template <typename T>
 void Fill(T* output, T value, int64_t count);
 
+/*
+  This is a utility wrapper for arbitrary type array
+  Commonly used for passing small list of metadata during cuda kernel launch
+  It's better to pass the array by value than having another cuMemcpy to pass the data to device.
+*/
+template <typename T, int32_t capacity = 8>
```

Comment:
Scott McKay
Any reason why default initialization of data_ can't be used to avoid the memset calls here?

Sherlock Huang
fixed